modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItThinClientCriteriaQueryTest.java

```diff
@@ -0,0 +1,119 @@
+import java.util.Map;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.lang.AsyncCursor;
+import org.apache.ignite.lang.Cursor;
+import org.apache.ignite.table.RecordView;
+import org.apache.ignite.table.Tuple;
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+/**
+ * Thin client criteria query test.
+ */
+@SuppressWarnings("resource")
+public class ItThinClientCriteriaQueryTest extends ItAbstractThinClientTest {
+    /** {@inheritDoc} */
+    @Override
+    @BeforeAll
+    void beforeAll(TestInfo testInfo, @WorkDirectory Path workDir) throws InterruptedException {
+        super.beforeAll(testInfo, workDir);
+
+        populateData(client(), TABLE_NAME);
+    }
+
+    @Test
+    public void testBasicQueryCriteriaRecordBinaryView() {
+        RecordView<Tuple> view = client().tables().table(TABLE_NAME).recordView();
+
+        try (Cursor<Tuple> cur = view.queryCriteria(null, null)) {
+            assertThat(Lists.newArrayList(cur), containsInAnyOrder(
+                    tupleValue(COLUMN_KEY, is(0)),
+                    tupleValue(COLUMN_KEY, is(1)),
+                    tupleValue(COLUMN_KEY, is(2))
+            ));
+        }
+
+        AsyncCursor<Tuple> ars = await(view.queryCriteriaAsync(null, null, builder().pageSize(2).build()));
+
+        assertNotNull(ars);
+        assertEquals(2, ars.currentPageSize());
+        await(ars.closeAsync());
+    }
+
+    @Test
+    public void testBasicQueryCriteriaRecordPojoView() {
+        RecordView<TestPojo> view = client().tables().table(TABLE_NAME).recordView(TestPojo.class);
+
+        try (Cursor<TestPojo> cur = view.queryCriteria(null, null)) {
+            assertThat(Lists.newArrayList(cur), containsInAnyOrder(
+                    hasProperty(COLUMN_KEY, is(0)),
+                    hasProperty(COLUMN_KEY, is(1)),
+                    hasProperty(COLUMN_KEY, is(2))
+            ));
+        }
+
+        AsyncCursor<TestPojo> ars = await(view.queryCriteriaAsync(null, null, builder().pageSize(2).build()));
+
+        assertNotNull(ars);
+        assertEquals(2, ars.currentPageSize());
+        await(ars.closeAsync());
+    }
+
+    private static void populateData(Ignite ignite, String tableName) {
+        RecordView<Tuple> table = ignite.tables().table(tableName).recordView();
+
+        for (int val = 0; val < 3; val++) {
+            table.insert(null, Tuple.create(Map.of(COLUMN_KEY, val % 100, "val", "some string row" + val)));
+        }
+    }
+
+    /**
+     * Creates a matcher for matching tuple value.
+     *
+     * @param valueMatcher Matcher for matching tuple value.
+     * @return Matcher for matching tuple value.
+     */
+    private static <T> Matcher<Tuple> tupleValue(String columnName, Matcher<T> valueMatcher) {
```

Comment:
we have two identical matchers. Propose to extract it to separate class to have ability reuse.
For example we have org.apache.ignite.internal.testframework.matchers

Reply:
Created `TupleMatcher`